docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,238 @@
+
+**[2]:** Determined by using the first of the following that applies:
+
+- Host identifier of the [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) if it's sent in absolute-form.
+- Host identifier of the `Host` header.
+
+**[3]:** When [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) is an absolute URI, `server.port` MUST match the URI port identifier; otherwise, it MUST match the `Host` header port identifier.
+
+**[4]:** If not the default (`80` for `http` scheme, `443` for `https`).
+<!-- endsemconv -->
+
+### Metric: `http.client.request.time_in_queue`
+
+<!-- semconv metric.dotnet.http.client.request.time_in_queue(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `http.client.request.time_in_queue` | Histogram | `s` | The amount of time requests spent on a queue waiting for an available connection. [1] |
+
+**[1]:** Meter name: `System.Net.Http`; Added in: .NET 8.0
+<!-- endsemconv -->
+
+<!-- semconv metric.dotnet.http.client.request.time_in_queue(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`http.request.method`](../attributes-registry/http.md) | string | HTTP request method. [1] | `GET`; `POST`; `HEAD` | Recommended |
+| [`network.protocol.version`](../attributes-registry/network.md) | string | Version of the protocol specified in `network.protocol.name`. [2] | `1.1`; `2` | Recommended |
+| [`server.address`](../general/attributes.md) | string | Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) the HTTP request is sent to. [3] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | Required |
+| [`server.port`](../general/attributes.md) | int | Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) the HTTP request is sent to. [4] | `80`; `8080`; `443` | Conditionally Required: [5] |
+| [`url.scheme`](../attributes-registry/url.md) | string | The [URI scheme](https://www.rfc-editor.org/rfc/rfc3986#section-3.1) component identifying the used protocol. | `http`; `https`; `ftp` | Recommended |
+
+**[1]:** HTTP request method value is one of the "known" methods listed in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods) and the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+If the HTTP request method isn't known, it sets the `http.request.method` attribute to `_OTHER`.
+It's not possible at the moment to override the list of known HTTP methods.
+
+**[2]:** `network.protocol.version` refers to the version of the protocol used and might be different from the protocol client's version. If the HTTP client has a version of `0.27.2`, but sends HTTP version `1.1`, this attribute should be set to `1.1`.
```

Comment:
Same as in the related comment on `http.client.open_connections`.

Reply:
I wonder if it should be the case here since` request.time_in_queue` is per-request metrics and not the connection one.

It should also be possible to return `HttpRequestMessage.Version` there since there is a request object.
 https://github.com/dotnet/runtime/blob/7990c077668117a74e967fac2afa5f7f6458ec07/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/SocketsHttpHandlerMetrics.cs#L35

I know it's too late to change in .NET 8, but WDYT about changing it at some point in the future?

Reply:
We discussed this topic in a code review and conciously decided to go with the current behavior -- see https://github.com/dotnet/runtime/pull/88981#discussion_r1265327139.

Even though this is strictly-speaking a request metric, it tells more about the state of the queues/connection pools than about the requests. This logic is also reflected in the EventCounter counterpart of this metric, where we have 3 EventCounters, one per each queue, see https://learn.microsoft.com/en-us/dotnet/core/diagnostics/available-counters#systemnethttp-counters.

Moreover, given that HTTP/1.0 is a niche legacy use-case, it's very unlikely we will spend efforts to change the current behavior.

Reply:
thanks for the explanation, I updated the metric definition.